src/services/WadValidationService.ts

```diff
@@ -0,0 +1,182 @@
+        const retVal = new WadValidationModel();
+
+        const allowedHeaders = await this.settingsService.getSetting(SETTING.ALLOWED_HEADERS);
+        const allowedFiles = await this.settingsService.getSetting(SETTING.ALLOWED_FILES);
+        const allowedFilesZip = await this.settingsService.getSetting(SETTING.ALLOWED_FILES_ZIP);
+        const allowedHeadersZip = await this.settingsService.getSetting(SETTING.ALLOWED_HEADERS_ZIP);
+
+        if (allowedHeaders) {
+            retVal.allowedHeaders = allowedHeaders.split(",")
+                .map(header => encodeURIComponent(header))
+                .map(ext => ext === "null" ? null : ext);
+        }
+        if (allowedFiles) {
+            retVal.allowedExtensions = allowedFiles.split(",");
+        }
+        if (allowedFilesZip) {
+            retVal.allowedExtensionsZip = allowedFilesZip.split(",");
+        }
+        if (allowedHeadersZip) {
+            retVal.allowedHeadersZip = allowedHeadersZip.split(",")
+                .map(header => encodeURIComponent(header))
+                .map(ext => ext === "null" ? null : ext);
+        }
+        return retVal;
+    }
+
+    public async validateWad(customWad: PlatformMulterFile): Promise<void> {
+        const [filesMap, filesZipMap] = await this.getMappings();
+        this.allowedFilesMap = filesMap;
+        this.allowedFilesZipMap = filesZipMap;
+
+        this.checkFileExt(customWad, false);
+        const fileExt = this.getFileExt(customWad);
+        if (fileExt === "ZIP") {
+            await this.analyseZip(customWad);
+        }
+        const buffer = await fs.promises.readFile(customWad.path);
+        this.checkHeaders(buffer, false, fileExt);
+    }
+
+    private async getMappings(): Promise<[Map<string, string | null>, Map<string, string | null>]> {
+        const mappings = await this.getHeaderMapping(false);
+        const zipMappings = await this.getHeaderMapping(true);
+        return [mappings, zipMappings];
+    }
+
+    private async getHeaderMapping(zip: boolean): Promise<Map<string, string | null>> {
+        let extensions: string | null;
+        if (zip) {
+            extensions = await this.settingsService.getSetting(SETTING.ALLOWED_FILES_ZIP);
+        } else {
+            extensions = await this.settingsService.getSetting(SETTING.ALLOWED_FILES);
+        }
+        if (!extensions) {
+            return new Map();
+        }
+        let headers: string | null;
+        if (zip) {
+            headers = await this.settingsService.getSetting(SETTING.ALLOWED_HEADERS_ZIP);
+        } else {
+            headers = await this.settingsService.getSetting(SETTING.ALLOWED_HEADERS);
+        }
+        return this.mapExtensions(extensions, headers);
+    }
+
+    private mapExtensions(extensions: string, headers: string | null): Map<string, string | null> {
+        const retMap: Map<string, string | null> = new Map();
+        const extensionsArr = extensions.split(",").map(f => f.toUpperCase());
```

Comment:
minor but `extensions.toUpperCase().split(",")` doesn't require an extra call to map.

Reply:
Good point